src/rtextures.c

```diff
@@ -1630,6 +1630,241 @@ Image ImageTextEx(Font font, const char *text, float fontSize, float spacing, Co
     return imText;
 }
 
+Image ImageFromChannel(Image image, int selected_channel, float threshold)
+{
+    Image result = { 0 };
+
+    // Security check to avoid program crash
+    if ((image.data == NULL) || (image.width == 0) || (image.height == 0))
+        return result;
+
+    // Check thresholds limits
+    if (threshold < 0.0f)
+    {
+        TRACELOG(LOG_WARNING, "Threshold is from 0 to 1, changing to 0.");
+        threshold = 0.0f;
+    }
+    if (threshold > 1.0f)
+    {
+        TRACELOG(LOG_WARNING, "Threshold is from 0 to 1, changing to 1.");
+        threshold = 1.0f;
+    }
+
+    // Check selected channel
+    if (image.format == PIXELFORMAT_UNCOMPRESSED_GRAYSCALE
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R32
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R16
+            )
+    {
+        if (selected_channel > 0)
+        {
+            TRACELOG(LOG_WARNING, "This image has only 1 channel. Setting channel to it.");
+            selected_channel = 0;
+        }
+    }
+    else if (image.format == PIXELFORMAT_UNCOMPRESSED_GRAY_ALPHA)
+    {
+        if (selected_channel > 1)
+        {
+            TRACELOG(LOG_WARNING, "This image has only 2 channels. Setting channel to alpha.");
+            selected_channel = 1;
+        }
+    }
+    else if (image.format == PIXELFORMAT_UNCOMPRESSED_R5G6B5
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R8G8B8
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R32G32B32
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R16G16B16
+            )
+    {
+        if (selected_channel > 2)
+        {
+            TRACELOG(LOG_WARNING, "This image has only 3 channels. Setting channel to red.");
+            selected_channel = 0;
+        }
+    }
+
+    // formats rgba
+    if (selected_channel > 3 || selected_channel < 0)
+    {
+        TRACELOG(LOG_WARNING, "ImageFromChannel supports channels 0 to 3 (rgba). Setting channel to alpha.");
+        selected_channel = 3;
+    }
+
+    result.format = PIXELFORMAT_UNCOMPRESSED_GRAY_ALPHA;
+    result.height = image.height;
+    result.width = image.width;
+    result.mipmaps = 1;
+
+    unsigned char *pixels = (unsigned char *)RL_CALLOC(image.width * 2 * image.height, sizeof(unsigned char)); // values 0 to 255
+
+    if (image.format >= PIXELFORMAT_COMPRESSED_DXT1_RGB) TRACELOG(LOG_WARNING, "IMAGE: Pixel data retrieval not supported for compressed image formats");
+    else
+    {
+        for (int i = 0, k = 0; i < image.width*2*image.height; i+=2)
+        {
+            float image_value = -1;
```

Comment:
Please, could you review [raylib naming conventions](https://github.com/raysan5/raylib/blob/master/CONVENTIONS.md)